unix/qstrdefsport.h

```diff
@@ -79,6 +79,7 @@ Q(listen)
 Q(accept)
 Q(recv)
 Q(recvfrom)
+Q(settimeout)
```

Comment:
> So the pre-processor does multiple passes? First, on this header file and then on the generated header file? 

well yes, but the generated header file does not contain any ifdefs anymore, only the qstr definitions

> I have no idea of what the implications of that are on the final binaries

just that it will be smaller in size, afaik

> Then what should be done about this?

for instance unix/qstrdefs.h can be improved by surrounding all socket related qstrs with ```#if MICROPY_PY_SOCKET```, the ffi related qstrs with ```#if MICROPY_PY_FFI``` and so on